lfs.c

```diff
@@ -4404,23 +4430,16 @@ static int lfs_mount_(lfs_t *lfs, const struct lfs_config *cfg) {
 
     // scan directory blocks for superblock and any global updates
     lfs_mdir_t dir = {.tail = {0, 1}};
-    lfs_block_t tortoise[2] = {LFS_BLOCK_NULL, LFS_BLOCK_NULL};
-    lfs_size_t tortoise_i = 1;
-    lfs_size_t tortoise_period = 1;
     while (!lfs_pair_isnull(dir.tail)) {
-        // detect cycles with Brent's algorithm
-        if (lfs_pair_issync(dir.tail, tortoise)) {
-            LFS_WARN("Cycle detected in tail list");
-            err = LFS_ERR_CORRUPT;
+        struct lfs_tortoise_t tortoise = {
+            .pair = {LFS_BLOCK_NULL, LFS_BLOCK_NULL},
+            .i = 1,
+            .period = 1,
+        };
```

Comment:
This needs to go outside the loop. Similar for the other 3 places.

Reply:
o(╥﹏╥)o